src/utils/rootkeypackage_utils/inc/aduc/rootkeypackage_types.h

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file rootkeypackage_types.h
+ * @brief rootkeypackage types.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#ifndef ROOTKEYPACKAGE_TYPES_H
+#define ROOTKEYPACKAGE_TYPES_H
+
+#include <azure_c_shared_utility/constbuffer.h>
+#include <azure_c_shared_utility/strings.h>
+#include <azure_c_shared_utility/vector.h>
+#include <sys/time.h> // for struct timespec
+
+/**
+ * @brief The root key keytype.
+ */
+typedef enum tagADUC_RootKey_KeyType
+{
+    RSA, /**< RSA keytype. */
+} ADUC_RootKey_KeyType;
+
+/**
+ * @brief The root key SHA algorithms.
+ */
+typedef enum tagADUC_RootKeyShaAlgorithm
+{
+    SHA256, /**< The SHA256 algorithm. */
+    SHA384, /**< The SHA384 algorithm. */
+    SHA512, /**< The SHA512 algorithm. */
+} ADUC_RootKeyShaAlgorithm;
```

Comment:
We should check with Dave/PRSS to make sure we're not planning on moving to 1024/2048+